### PR TITLE
feat: add basic support for nRF51 chips to embassy-nrf

### PR DESCRIPTION
**Pull request by lulf**
TODO:

* [ ] HIL test
* [ ] Docs update
* [ ] More peripherals (can be added on demand later I would say)

@Dirbaio would you be able to hook your nrf51-dk into the HIL setup? My own dk is fried, but I verified code is running on my microbit v1.

### CODE REVIEW

File: examples/nrf51/.cargo/config.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[target.'cfg(all(target_arch = "arm", target_os = "none"))']
+# replace nRF82840_xxAA with your chip as listed in `probe-rs chip list`
```
Comment (plaes) [inline examples/nrf51/.cargo/config.toml:2]: Not sure whether nrf52840 makes sense here?

**Comment by Dirbaio** [comment]
I think we could (should?) avoid adding the dependency on portable-atomic, 
- RNG could use a `Mutex<RefCell<State>>` (and change the AtomicWaker to a WakerRegistration). It'd be faster because if you modify multiple fields it'll acquire a single CS, not one for each field.
- The time driver already does a CS for `next_period` so we can do "read, modify, write" manually which needs no CAS.

**Comment by Dirbaio** [comment]
added nrf51-dk to HIL rig. Wire connected between P0.13 and P0.14 for GPIO etc testing.

**Comment by lulf** [comment]
@Dirbaio Thanks! I'll add a HIL test as well as addressing the comments soon!

**Comment by lulf** [comment]
bender run

**Comment by lulf** [comment]
@Dirbaio Two issues remaining:

* Running from RAM doesn't work, it looks related to the size and the available RAM because a simple test that doesn't do gpio or timer runs fine from RAM. 
* I've set it to run from flash, which works locally, but fails with `ERROR - Run failed: Flash content verification failed.` I'm not entirely sure why this is only happening on CI.

**Comment by Dirbaio** [comment]
> Running from RAM doesn't work, it looks related to the size and the available RAM because a simple test that doesn't do gpio or timer runs fine from RAM.

that's fine. we already have some stm32's that do HIL from flash because not enough RAM too...

> I've set it to run from flash, which works locally, but fails with ERROR - Run failed: Flash content verification failed. I'm not entirely sure why this is only happening on CI.

Fixed in https://github.com/probe-rs/probe-rs/pull/2119

bender run

**Comment by Dirbaio** [comment]
duh of course it doesn't work if I don't put back the nrf51-dk in the HL rig.

bender run

**Review by Dirbaio — APPROVED** [review]
thank you! :partying_face:

**Pull request merged**